### DISCUSSION TITLE
Raspberry Pi 4: Can I run TTS and STT in (faster than) real-time?

### DISCUSSION THREAD

**Discussion by repodiac** [coqui-ai/TTS#904] [General Q&A]
Hi awesome people,

I am currently looking into building a physical voice assistant device based on, for instance your TTS and STT framework.

Therefore, I would like to know if I can use already the current release(s) of TTS and STT for a real-time voice assistant, i.e. talking to the device is as close as possible to commercial devices like echo or alexa in terms of processing speech (not to mention the "intelligent" backend part, but that is not part of my question/concern at first).

I found various issues/posts on the repo from the past (talking about deepspeech and/or mozilla TTS), but that seems a bit outdated or I am not sure if I can rely on those "early" results?

Since you prominently claim on your corporate site, that both STT and TTS run from raspberry pi 4 to a full server with gpus I wonder if this claim is justified or if there are any snags or something in "the small print" somewhere...?

Thanks in advance for any clarifications here!

**Comment by erogol** [comment]
I know some people run certain 🐸TTS models on Raspi with success but I've not tried personally.

For STT @reuben would know better.

**Reply by reuben** [reply]
STT can transcribe faster than real time on a RPi3B+ or faster.

**Reply by repodiac** [reply]
Thanks, if that's all to know about.
